src/compiler/diagnosticMessages.json

```
@@ -3264,6 +3268,10 @@
         "category": "Error",
         "code": 2802
     },
+    "Import assertions are only supported when the '--module' flag is set to 'esnext'.": {

```

```suggestion
    "Import assertions are only supported when the '--module' option is set to 'esnext'.": {
```

It's not a "flag" (in that its not a boolean value). Most similar diagnostic messages use `option is set to`, with the exception of TS1323 (which arguably should be changed to match the other messages).